### PR TITLE
TB_Airport

### PR DESCRIPTION
**Pull request by TBrockster**
Finished user stories, but not refactored, using doubles or tested for edge-cases.

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  attr_reader :local_weather
+  attr_reader :capacity
+  attr_reader :airport_id
+  attr_writer :local_weather
+  DEFAULT_CAPACITY = 5
+
+  def initialize(airport_id: "Toby's Airport", capacity: DEFAULT_CAPACITY)
+    @planes = []
+    @airport_weather_object = Weather.new
+    @local_weather = @airport_weather_object.weather
```
Comment (alicelieutier) [inline lib/airport.rb]: What does this line do?

File: lib/weather.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Weather
+  attr_reader :weather
+  def initialize
+    generate_weather
+  end
+  
+  def generate_weather
+    rand(5).zero? ? @weather = 'Stormy' : @weather = 'Sunny'
```
Comment (alicelieutier) [inline lib/weather.rb]: Why are you using a ternary expression here? Could this be more readable in two lines?

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
+class Airport
```
Comment (ckettell) [inline lib/airport.rb]: I may be wrong, but I think you may need require relative at the top. I tried loading just airport.rb in IRB and meant I couldn't create an instance.

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  attr_reader :local_weather
+  attr_reader :capacity
+  attr_reader :airport_id
+  attr_writer :local_weather
+  DEFAULT_CAPACITY = 5
+
+  def initialize(airport_id: "Toby's Airport", capacity: DEFAULT_CAPACITY)
+    @planes = []
+    @airport_weather_object = Weather.new
```
Comment (ted-scanlan) [inline lib/airport.rb]: I get why you have your own class, but for me generating weather in the Airport class would be slightly more readable

File: lib/airport.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  def full?
+    @planes.count >= capacity
+  end
+
+  def stormy?
+    @local_weather == 'Stormy'
+  end
+
+  def flying?(plane)
```
Comment (ckettell) [inline lib/airport.rb]: This method I do not think works. On line 19, you call plane.flying? Whereas, this one requires an argument. Means that I can't land a plane

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    @airport_weather_object = Weather.new
+    @local_weather = @airport_weather_object.weather
+    @capacity = capacity
+    @airport_id = airport_id
+  end
+
+  def land_plane(plane)
+    fail 'The Stormy weather prevents landing' if stormy?
+    fail 'The Airport is full' if full?
+    fail 'A plane cannot land if it is not flying' unless plane.flying?
```
Comment (ted-scanlan) [inline lib/airport.rb]: would this create a more simple design if you were to use the state attr_reader in the Plane class instead?

File: lib/airport.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    plane.state = @airport_id
+    @planes << plane
+    plane
+  end
+
+  def takeoff_plane
```
Comment (ckettell) [inline lib/airport.rb]: Without taking an argument, you wouldn't be able to find out if a specific plane has left the airport. By using pop, you can't keep track of which ones have left

File: spec/airport_spec.rb [hunk 8]
```diff
@@ -0,0 +1,82 @@
+require 'airport'
+
+describe Airport do
+  it { is_expected.to respond_to(:land_plane).with(1).argument }
+  it { is_expected.to respond_to(:takeoff_plane) }
```
Comment (ted-scanlan) [inline spec/airport_spec.rb]: these "respond_to" are not that necessary

File: spec/airport_spec.rb [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  it { is_expected.to respond_to(:regenerate_weather) }
+  let(:flying_plane) { double :plane, :state => 'Flying', :state= => :state, :flying? => :state }
+  let(:stored_plane) { double :plane, :state => subject.airport_id, :state= => :state, :flying? => :state }
```
Comment (ted-scanlan) [inline spec/airport_spec.rb]: I'm not sure what's going on in the second half of this line

File: spec/airport_spec.rb [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+  it 'has a variable capacity' do
+    my_airport = Airport.new(capacity: 10)
+    my_airport.local_weather = 'Sunny'
+    10.times { my_airport.land_plane(flying_plane) }
+    expect { my_airport.land_plane(flying_plane) }.to raise_error 'The Airport is full'
+  end
+
+  describe '#land_plane' do
+    it 'stores planes' do
+      subject.local_weather = 'Sunny'
```
Comment (ted-scanlan) [inline spec/airport_spec.rb]: it is better to use a stub here to replace the random i.e. allow(subject.to receive(class method) {value}

**Pull request closed without merging**